### ISSUE TITLE
Skeleton should reflect the exact overloaded called method

### ISSUE THREAD

**Issue by HDouss** [yegor256/jpeek#403]
Today, the skeleton reflects method calls by their name. For example, the skeleton for [OverloadMethods.java](https://github.com/yegor256/jpeek/blob/master/src/test/resources/org/jpeek/samples/OverloadMethods.java) have:
```xml
               <method abstract="false"
                       bridge="false"
                       ctor="false"
                       desc="(Ljava/lang/String;)D"
                       name="methodOne"
                       static="false"
                       visibility="public">
                  <args>
                     <arg type="Ljava/lang/String">?</arg>
                  </args>
                  <return>D</return>
                  <ops>
                     <op code="get">num</op>
                     <op code="put">num</op>
                     <op code="call">OverloadMethods.methodOne</op>
                  </ops>
               </method>
               <method abstract="false"
                       bridge="false"
                       ctor="false"
                       desc="(Ljava/lang/String;Ljava/lang/String;)D"
                       name="methodOne"
                       static="false"
                       visibility="public">
                  <args>
                     <arg type="Ljava/lang/String">?</arg>
                     <arg type="Ljava/lang/String">?</arg>
                  </args>
                  <return>D</return>
                  <ops>
                     <op code="get">num</op>
                     <op code="put">num</op>
                     <op code="call">OverloadMethods.methodOne</op>
                  </ops>
               </method>
```

The two calls are reported in `<op code="call">` elements are `OverloadMethods.methodOne` and no information is given on which overloaded method is called. This is specially important to calculate LCOM4.

**Comment by 0crat** [comment]
@paulodamaso[/z](https://www.0crat.com/u/paulodamaso) please, pay attention to this issue

**Comment by paulodamaso** [comment]
@HDouss You are proposing that we should add this information to skeleton?

**Comment by HDouss** [comment]
@paulodamaso yes this information is clearly lacking. See the example in the ticket description. Looking to the skeleton, we don't know which method is called, we just know that it is called `methodOne`, but there are 4 `methodOne`s in the class.

**Comment by paulodamaso** [comment]
@HDouss Okay, thanks for the clarification

**Comment by paulodamaso** [comment]
@0crat in

**Comment by 0crat** [comment]
The job [#403](https://github.com/yegor256/jpeek/issues/403) assigned to @dgarus[/z](https://www.0crat.com/u/dgarus), here is [why](https://www.0crat.com/footprint/C7JGJ00DP/ed15ac33-eb1c-4e3a-8f14-cec24b039036); the budget is 30 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§8](http://www.zerocracy.com/policy.html#8) and [§9](http://www.zerocracy.com/policy.html#9); if the task is not clear, read [this](http://www.yegor256.com/2015/02/16/it-is-not-a-school.html) and [this](http://www.yegor256.com/2015/01/15/how-to-cut-corners.html); there will be a monetary reward for this job

**Comment by 0crat** [comment]
@dgarus[/z](https://www.0crat.com/u/dgarus) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by dgarus** [comment]
@HDouss @paulodamaso 
> @HDouss You are proposing that we should add this information to skeleton?

I've read the issue and links double and didn`t find what is _this information to skeleton_
Should I propose a new structure of <op code="call"> tag or that structure already exists?

**Comment by dgarus** [comment]
@0crat wait

**Comment by HDouss** [comment]
@dgarus When analyzing a class and calculating its metric, jpeek starts by building a skeleton. The skeleton is an XML representation of the class, its method, its attributes, and the instructions in each method. For now, the skeleton produced contains the method calls instructions. These method calls instructions contains only the name of the called method. This is not enough to know which method is actually called, because many methods could have the same name.

**Comment by dgarus** [comment]
@HDouss  I got it but I asked 
>_we should add t**his information** to skeleton_

"**_This information_**" is already described or I should propose a new structure to describe this information?

**Comment by HDouss** [comment]
> @HDouss I got it but I asked
> 
> > _we should add t**his information** to skeleton_
> 
> "**_This information_**" is already described or I should propose a new structure to describe this information?

@dgarus It is not described. The most important thing is that it exists. I could suggest this:
```xml
<op code="call">OverloadMethods.methodOne(Ljava/lang/String;Ljava/lang/String;)D</op>
```

or

```xml
<op code="call">
    <name>OverloadMethods.methodOne</name>
    <signature>(Ljava/lang/String;Ljava/lang/String;)D</signature>
</op>
```

**Comment by dgarus** [comment]
@HDouss Ok, I got it. Thx!
I like your second variant.
I thought about this way
```
<op code="call">
    <name>OverloadMethods.methodOne</name>
    <args>
    	<arg type="Ljava/lang/String">?</arg>
    	<arg type="Ljava/lang/String">?</arg>
    </args>
</op>
```

It seems more formal. WDYT?

**Comment by HDouss** [comment]
@dgarus agree. Note that this will have a rather big impact on the current metrics calculations. @paulodamaso What do you think?

**Comment by paulodamaso** [comment]
@HDouss I think that it would be a good idea. Which metrics will be affected by this change?

**Comment by HDouss** [comment]
@paulodamaso it isn't the metrics, but the way to calculate them. Nearly every xsl file should be modified to adapt its xpath that selects called methods.

**Comment by 0pdd** [comment]
@HDouss the puzzle [#437](https://github.com/yegor256/jpeek/issues/437) is still not solved.

**Comment by dgarus** [comment]
@HDouss could you please close the issue?

**Comment by HDouss** [comment]
@dgarus Thanks

**Comment by 0crat** [comment]
@sereshqua[/z](https://www.0crat.com/u/sereshqua) please review this job completed by @dgarus[/z](https://www.0crat.com/u/dgarus), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by sereshqua** [comment]
@0crat quality good

**Comment by 0pdd** [comment]
@HDouss 2 puzzles [#469](https://github.com/yegor256/jpeek/issues/469), [#470](https://github.com/yegor256/jpeek/issues/470) are still not solved; solved: [#437](https://github.com/yegor256/jpeek/issues/437).

**Comment by 0pdd** [comment]
@HDouss the puzzle [#469](https://github.com/yegor256/jpeek/issues/469) is still not solved; solved: [#437](https://github.com/yegor256/jpeek/issues/437), [#470](https://github.com/yegor256/jpeek/issues/470).
